neural_compressor/torch/algorithms/weight_only/gptq.py

```diff
@@ -542,8 +545,12 @@ def forward(layer, *args, **kwargs):
         if self.run_fn:
             if self.run_args:
                 self.run_fn(self.model, *self.run_args)
+                if "hpu" in self.device:

```

Comment:
How about adding `mark_step` into [Auto_Accelerator](https://github.com/intel/neural-compressor/blob/24419c9044fe227ea806db370c1a30272d026f8a/neural_compressor/torch/utils/auto_accelerator.py#L208)?
We could then directly use `auto_detect_accelerator().mark_step()` without needing to verify the device each time,  also make the code cleaner?

I've applied this approach with `empty_cache()` at HQQ.
https://github.com/intel/neural-compressor/blob/24419c9044fe227ea806db370c1a30272d026f8a/neural_compressor/torch/algorithms/weight_only/hqq/core.py#L156-L157

Reply:
done